zulip_bots/zulip_bots/bots/googletranslate/googletranslate.py

```diff
@@ -0,0 +1,92 @@
+# To use this plugin, you need to set up the Google Cloud API key for this bot in
+# googletranslate.conf in this (zulip_bots/bots/googletranslate/) directory.
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+from zulip_bots.bots.googletranslate import utils
+import requests
+
+class GoogleTranslateHandler(object):
+    '''
+    This bot will translate any messages sent to it using google translate.
+    Before using it, make sure you set up google api keys, and enable google
+    cloud translate from the google cloud console.
+    '''
+    def usage(self):
+        return '''
+            This plugin allows users translate messages
+            Users should @-mention the bot with the format
+            @-mention "<text>" <target-language> <source-language(optional)>
+            '''
+
+    def initialize(self, bot_handler):
+        self.config_info = bot_handler.get_config_info('googletranslate')
+
+    def handle_message(self, message, bot_handler):
+        bot_response = get_translate_bot_response(message['content'], self.config_info, message['sender_full_name'])
+        bot_handler.send_reply(message, bot_response)
+
+api_url = 'https://translation.googleapis.com/language/translate/v2'
+
+help_text = '''
+Google translate bot
+Please format your message like:
+`@-mention "<text>" <target-language> <source-language(optional)>`
+To see a list of languages, @-mention this bot with `languages`
+'''
+
+languages_text = '''Please see [here](https://cloud.google.com/translate/docs/languages) for all languages and their codes.
+You can use these codes, or their names.'''
+
+class TranslateError(Exception):
+    def __init__(self, value):
+        self.value = value
+
+    def __str__(self):
+        return repr(self.value)
+
+def translate(text, key, dest, src):
+    parameters = {'q': text, 'target': dest, 'key': key}
+    if (src != ''):
+        parameters.update({'source': src})
+    r = requests.post(api_url, params=parameters)
+    if (r.status_code == requests.codes.ok):
+        return r.json()['data']['translations'][0]['translatedText']
+    raise TranslateError(r.json()['error']['message'])
+
+def get_translate_bot_response(text, config_file, author):
+    text = text.strip()
+    if (text == 'languages'):
+        return languages_text
+    if (text == 'help' or text is None or not text.startswith('"')):
+        # Return the help message
+        return help_text
+    split_text = []
+    split_text = text.split('" ')
+    if (len(split_text) == 1):
+        return help_text
+    split_text += split_text.pop(1).split(' ')
+    if (len(split_text) == 2):
+        # There is no source language
+        split_text.append("")
+    if (len(split_text) != 3):
+        return help_text
+    split_text[0] = split_text[0].replace('"', '')
+    if (not split_text[1] in utils.google_langs.keys()):
```

Comment:
If you want to refactor `utils.py` then you could consider making this a short function, to allow the `google_langs` to be varied.